coaster/sqlalchemy.py

```diff
@@ -95,13 +95,42 @@ def permissions(self, user, inherited=None):
 
 class UrlForMixin(object):
     """
-    Provides a placeholder :meth:`url_for` method used by BaseMixin-derived classes
+    Provides a :meth:`url_for` method used by BaseMixin-derived classes
     """
+    #: Mapping of {action: (endpoint, {param: attr})}, where attr is a string or tuple of strings.
+    #: This particular dictionary is only used as a fallback. Each subclass will get its own dictionary.
+    url_for_endpoints = {}
+
     def url_for(self, action='view', **kwargs):
         """
         Return public URL to this instance for a given action (default 'view')
         """
-        return None
+        if action not in self.url_for_endpoints:
+            return  # XXX: Legacy behaviour, fail silently
+        endpoint, paramattrs = self.url_for_endpoints[action]
+        for param, attr in paramattrs.items():
+            if isinstance(attr, tuple):
+                item = self
+                for subattr in attr:
+                    item = getattr(item, subattr)
+                kwargs[param] = item
+            else:
+                kwargs[param] = getattr(self, attr)
+
+        return url_for(endpoint, **kwargs)
```

Comment:
Could this could use a more obvious name to indicate that it's calling flask's `url_for`? At first blush, it seemed like a recursive call.